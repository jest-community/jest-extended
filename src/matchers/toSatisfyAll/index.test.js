import matcher from './';

expect.extend(matcher);

let isEven = el => el % 2 === 0;
let isOdd = el => el % 2 === 1;

describe('.toSatisfyAll', () => {
  it('passes when all values satisfy predicate', () => {
    expect([1, 3, 5, 7]).toSatisfyAll(isOdd);
    expect([2, 4, 6, 8]).toSatisfyAll(isEven);
    expect([11]).toSatisfyAll(isOdd);
    expect([10]).toSatisfyAll(isEven);
  });

  it('fails when any value does not satisfy the predicate', () => {
    expect(() => expect([1, 3, 4, 5]).toSatisfyAll(isOdd)).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toSatisfyAll', () => {
  it('passes when any value does not satisfy the predicate', () => {
    expect([1, 3, 4, 5]).not.toSatisfyAll(isOdd);
    expect([8, 6, 3, 1, 2]).not.toSatisfyAll(isEven);
  });

  it('fails when all values satisfy predicate', () => {
    expect(() => expect([1, 3, 5, 7]).not.toSatisfyAll(isOdd)).toThrowErrorMatchingSnapshot();
  });
});
