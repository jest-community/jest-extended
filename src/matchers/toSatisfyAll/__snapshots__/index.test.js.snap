// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toSatisfyAll fails when all values satisfy predicate 1`] = `
"<dim>expect(</><red>received</><dim>).not.toSatisfyAll(</><green>expected</><dim>)</>

Expected array to not satisfy predicate for all values:
  <green>[Function isOdd]</>
Received:
  <red>[1, 3, 5, 7]</>"
`;

exports[`.toSatisfyAll fails when any value does not satisfy the predicate 1`] = `
"<dim>expect(</><red>received</><dim>).toSatisfyAll(</><green>expected</><dim>)</>

Expected array to satisfy predicate for all values:
  <green>[Function isOdd]</>
Received:
  <red>[1, 3, 4, 5]</>"
`;
